packages/test-runner/src/dispatcher.ts

```diff
@@ -143,19 +146,27 @@ export class Dispatcher {
     await Promise.all(jobs);
   }
 
-  async _runJob(worker: Worker, entry: TestRunnerEntry) {
-    worker.run(entry);
+  async _runJob(worker: Worker, runPayload: RunPayload) {
+    for (const entry of runPayload.entries)
+      this._testOutputById.set(entry.testId, createTestResult(0));
+    worker.run(runPayload);
+
     let doneCallback;
     const result = new Promise(f => doneCallback = f);
-    worker.once('done', params => {
+    const done = () => {
```

Comment:
This makes understanding considerably harder, what are you trying to achieve here?